cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+
+import cfme.fixtures.pytest_selenium as sel
+from cfme import test_requirements
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.ec2 import EC2Provider
+from cfme.cloud.provider.gce import GCEProvider
+from cfme.web_ui import flash, form_buttons
+from utils import testgen, error
+from utils.conf import cfme_data, credentials
+from utils.log import logger
+from utils.appliance.implementations.ui import navigate_to
+
+pytestmark = [
+    pytest.mark.tier(1),
+    test_requirements.proxy,
+    pytest.mark.usefixtures('setup_provider_modscope')
+]
+pytest_generate_tests = testgen.generate([AzureProvider, GCEProvider, EC2Provider],
+                                         scope="module")
+
+
+def test_proxy_default(appliance, provider, request):
+    """Configures the specific valid proxy for each provider
+
+    Args:
```

Comment:
Whilst I agree here that the args are None, I think we are all aware that pytest tests are special cases, I would like to remove the args definitions here, we are never going to have args to these functions that a user can pass in.